### PR TITLE
Add dev.jonmagon.kdiskmark

### PR DESCRIPTION
**Pull request by JonMagon**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
- **`--filesystem=host`: Since this is a benchmarking application, it needs access to write to almost every mount point, including external storage devices.**
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- **Technically, the domain is not registered anywhere. The same namespace for DBus is used, reserved in SUSE.**
- [x] Any additional patches or files have been submitted to the upstream projects concerned.
- **The `legacy` branch is used for the sandboxed packages build. Since the `master` uses Polkit, it cannot be used here.**

Requested by issues: https://github.com/JonMagon/KDiskMark/issues/34, https://github.com/JonMagon/KDiskMark/issues/98

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
> * Technically, the domain is not registered anywhere. The same namespace for DBus is used, reserved in SUSE.

Then the app-id doesn't meet the requirements.

File: dev.jonmagon.kdiskmark.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+rename-desktop-file: kdiskmark.desktop
+rename-icon: kdiskmark
+copy-icon: true
+command: kdiskmark
+finish-args:
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline dev.jonmagon.kdiskmark.yaml:11]:
```suggestion
  - --share=ipc
  - --socket=fallback-x11
  - --socket=wayland
```

File: dev.jonmagon.kdiskmark.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    sources:
+      - type: git
+        url: https://github.com/JonMagon/KDiskMark.git
+        tag: 3.1.2-standalone
+        commit: b411c4ae0d4a4ab23531ae1fecddcf76a493a515
+        x-checker-data:
+          type: git
+    modules:
+      - name: fio
+        buildsystem: simple
```
Comment (hfiguiere) [inline dev.jonmagon.kdiskmark.yaml:29]: this should be `autotools`

**Comment by JonMagon** [comment]
> Then the app-id doesn't meet the requirements.

Maybe I misunderstood something, what *domain* are we talking about?

File: dev.jonmagon.kdiskmark.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            buildsystem: simple
+            build-commands:
+              - make prefix=$FLATPAK_DEST install
+            sources:
+              - type: git
+                url: https://pagure.io/libaio.git
+                tag: libaio-0.3.113
+                commit: 1b18bfafc6a2f7b9fa2c6be77a95afed8b7be448
+                x-checker-data:
+                type: git
```
Comment (hfiguiere) [inline dev.jonmagon.kdiskmark.yaml:52]:
```suggestion
                x-checker-data:
                    type: git
```

it's yaml, indentation is significant.

File: dev.jonmagon.kdiskmark.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+app-id: dev.jonmagon.kdiskmark
```
Comment (hfiguiere) [inline dev.jonmagon.kdiskmark.yaml:1]: as said earlier if the domain isn't registered then this doesn't meet the requirements.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build dev.jonmagon.kdiskmark

**Comment by hfiguiere** [comment]
not appstream file is found.

**Comment by JonMagon** [comment]
bot, build io.github.jonmagon.kdiskmark

**Comment by JonMagon** [comment]
fio has go away

**Comment by JonMagon** [comment]
bot, build io.github.jonmagon.kdiskmark

**Comment by JonMagon** [comment]
```
ERROR   src.manifest: Failed to check git libaio/libaio.git with GitChecker: https://pagure.io/libaio.git has no tags matching '^(?:[vV])?((?:\d+\.)+\d+)$': list index out of range
ERROR   src.manifest: Failed to check git fio/fio.git with GitChecker: https://github.com/axboe/fio.git has no tags matching '^(?:[vV])?((?:\d+\.)+\d+)$': list index out of range
```
What could be wrong?

**Comment by hfiguiere** [comment]
This:
image

**Comment by hfiguiere** [comment]
no.

**Comment by JonMagon** [comment]
bot, build io.github.jonmagon.kdiskmark

File: io.github.jonmagon.kdiskmark.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+rename-desktop-file: kdiskmark.desktop
+rename-icon: kdiskmark
+copy-icon: true
+command: kdiskmark
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=host
+  - --device=dri
```
Comment (hfiguiere) [inline io.github.jonmagon.kdiskmark.yaml:14]: you probably need to ensure you allow the dbus name, unless it got changed too to match the app-id.

File: io.github.jonmagon.kdiskmark.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+modules:
+  - name: kdiskmark
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
```
Comment (hfiguiere) [inline io.github.jonmagon.kdiskmark.yaml:19]:
```suggestion
      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

This will allow to have debug symbols.

**Comment by JonMagon** [comment]
It was merged, but I still can't use the built flatpak. The problem is in fio: `flatpak run --command=sh io.github.jonmagon.kdiskmark -c '/app/bin/fio'` does nothing. Probably need to add `config-opts: [ "--disable-native" ]`. PR is closed, how can I trigger yet another test build?
```bash
 %  ~  gdb ./fio                                                                                                                                                                                                                                    Вт 20 сен 2022 20:33:01
GNU gdb (GDB; openSUSE Tumbleweed) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-suse-linux".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://bugs.opensuse.org/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./fio...
Missing separate debuginfo for /home/jonmagon/fio
Try: zypper install -C "debuginfo(build-id)=c096b7c83e14f63d37f448770f27568ca544f58f"
(No debugging symbols found in ./fio)
(gdb) r
Starting program: /home/jonmagon/fio 
Missing separate debuginfos, use: zypper install glibc-debuginfo-2.36-4.1.x86_64
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
0x0000555555572530 in ?? ()
Missing separate debuginfos, use: zypper install krb5-debuginfo-1.20-1.2.x86_64 libaio1-debuginfo-0.3.113-1.2.x86_64 libbrotlicommon1-debuginfo-1.0.9-1.10.x86_64 libbrotlidec1-debuginfo-1.0.9-1.10.x86_64 libcom_err2-debuginfo-1.46.5-3.1.x86_64 libcurl4-debuginfo-7.85.0-1.1.x86_64 libidn2-0-debuginfo-2.3.3-1.2.x86_64 libkeyutils1-debuginfo-1.6.3-4.3.x86_64 libldap2-debuginfo-2.6.3-1.2.x86_64 libnghttp2-14-debuginfo-1.49.0-1.2.x86_64 libopenssl1_1-debuginfo-1.1.1q-1.3.x86_64 libpcre2-8-0-debuginfo-10.40-3.3.x86_64 libpsl5-debuginfo-0.21.1-1.13.x86_64 libsasl2-3-debuginfo-2.1.28-1.6.x86_64 libselinux1-debuginfo-3.4-2.3.x86_64 libssh4-debuginfo-0.10.4-1.1.x86_64 libunistring2-debuginfo-1.0-1.4.x86_64 libz1-debuginfo-1.2.12-1.1.x86_64 libzstd1-debuginfo-1.5.2-1.9.x86_64
```

**Pull request closed without merging**